node/sc/bridgepool/sorted_map_list.go

```diff
@@ -100,6 +100,23 @@ func (m *ItemSortedMap) Put(event itemWithNonce) {
 	m.mu.Lock()
 	defer m.mu.Unlock()
 
+	m.put(event)
+}
+
+// PutWithLimit inserts a new item into the map with size limit of the map.
+func (m *ItemSortedMap) PutWithLimit(event itemWithNonce, limit int) {
```

Comment:
If the new value is more suitable, do you have to delete the existing value and keep the new one?

Reply:
Old one is already in the map is more important than the new one.